locust/test/test_main.py

```diff
@@ -87,160 +111,197 @@ def test_custom_arguments(self):
     def test_custom_exit_code(self):
         with temporary_file(
             content=textwrap.dedent(
                 """
-            from locust import User, task, constant, events
-            @events.quitting.add_listener
-            def _(environment, **kw):
-                environment.process_exit_code = 42
-            @events.quit.add_listener
-            def _(exit_code, **kw):
-                print(f"Exit code in quit event {exit_code}")
-            class TestUser(User):
-                wait_time = constant(3)
-                @task
-                def my_task(self):
-                    print("running my_task()")
-        """
+                from locust import User, task, constant, events
+                @events.quitting.add_listener
+                def _(environment, **kw):
+                    environment.process_exit_code = 42
+                @events.quit.add_listener
+                def _(exit_code, **kw):
+                    print(f"Exit code in quit event {exit_code}")
+                class TestUser(User):
+                    wait_time = constant(3)
+                    @task
+                    def my_task(self):
+                        print("running my_task()")
+            """
             )
         ) as file_path:
-            proc = subprocess.Popen(["locust", "-f", file_path], stdout=PIPE, stderr=PIPE, text=True)
-            gevent.sleep(1)
-            proc.send_signal(signal.SIGTERM)
-            stdout, stderr = proc.communicate()
-            self.assertIn("Starting web interface at", stderr)
-            self.assertIn("Starting Locust", stderr)
-            self.assertIn("Shutting down (exit code 42)", stderr)
-            self.assertIn("Exit code in quit event 42", stdout)
-            self.assertEqual(42, proc.returncode)
+            port = get_free_tcp_port()
+            proc = subprocess.Popen(
+                ["locust", "-f", file_path, "--web-port", str(port)], stdout=PIPE, stderr=PIPE, text=True
+            )
+
+            try:
+                # Polling to ensure the process has started on the chosen port
+                poll_until(lambda: is_port_in_use(port), timeout=20)
+
+                proc.send_signal(signal.SIGTERM)
+                stdout, stderr = proc.communicate(timeout=3)
+
+                self.assertIn("Starting web interface at", stderr)
+                self.assertIn("Starting Locust", stderr)
+                self.assertIn("Shutting down (exit code 42)", stderr)
+                self.assertIn("Exit code in quit event 42", stdout)
+                self.assertEqual(42, proc.returncode)
+            except PollingTimeoutError:
+                self.fail(f"Failed to start Locust process on port {port}")
+            finally:
+                proc.terminate()
 
     @unittest.skipIf(os.name == "nt", reason="Signal handling on windows is hard")
     def test_webserver(self):
+        port = get_free_tcp_port()  # Use a free port and pass it to Locust
```

Comment:
Just a small nitpick here, I'm of the opinion that code should speak for itself. I think here and in a few other places we are repeating the same comment, and this comment doesn't really give any new information. `get_free_tcp_port` is already descriptive enough to tell us that we are getting a free port right?